client/client.go

```diff
@@ -141,21 +141,27 @@ func FromContext(ctx context.Context) Info {
 
 // NewMetadata creates a new Metadata object to use in Info. md is used as-is.
 func NewMetadata(md map[string][]string) Metadata {
+	c := make(map[string][]string, len(md))
+	for k, v := range md {
+		c[k] = v
```

Comment:
Let's use `strings.ToLower(k)` here and in get do a `m.data[strings.ToLower(k)]` then the optimization we removed is no longer needed.

Reply:
Done